plugin/goobook_complete.vim

```diff
@@ -46,8 +46,11 @@ function! goobook_complete#Complete(findstart, base)
 endfunc
 
 function! goobook_complete#Trim(res)
-    let trim="sed '/^$/d' | grep -v '(group)$' | cut -f1,2"
-    return split(system(trim, a:res), '\n')
+    let splits = split(a:res, '\r*\n')
+    let splits = filter(splits, 'v:val != ""')                 " Remove empty lines
+    let splits = filter(splits, 'v:val !~ "(other)$"')         " Remove lines ending with (other)
```

Comment:
What's this for? I added it because it was there already (e.g. `grep -v '(group)$') but...¯\_(ツ)_/¯

Reply:
AFAIR sometimes there are some lines which have suffixes like `(group)` or `(other)`, but I'd have to check if that is still the case. I'm also not sure if those lines should be filtered in any case.

With respect to the `\r*` I have a question, though. IMO it should really be a `\r?`, right? (Ignoring the case for older Mac OSes which had only `\r` but no `\n`).

Reply:
One more thing: I haven't played much with vimscript in a while, but we should maybe make sure to filter blank lines instead of only empty ones. What do you think?

Reply:
I just checked the output of `goobook` and I can confirm that it's necessary to remove lines ending with `(group)` which would otherwise reflect group tags from Google contacts, including all members of the respective groups. Not sure if filtering `(other)` is necessary, though.

Reply:
>With respect to the \r* I have a question, though. IMO it should really be a \r?, right? (Ignoring the case for older Mac OSes which had only \r but no \n).

Correct! I have been a little too sloppy with `\r*`

Reply:
> One more thing: I haven't played much with vimscript in a while, but we should maybe make sure to filter blank lines instead of only empty ones. What do you think?

What do you mean by _blank lines_? Lines with a bunch of whitespaces? I guess we could, but to be honest I focused on porting what we had before to Vimscript.

Reply:
> I just checked the output of `goobook` and I can confirm that it's necessary to remove lines ending with `(group)` which would otherwise reflect group tags from Google contacts, including all members of the respective groups. Not sure if filtering `(other)` is necessary, though.

Just to be on the safe side, shouldn't we list them all (i.e. without additional filters) and let users figure out the exact contact they are looking for?

Reply:
> Correct! I have been a little too sloppy with `\r*`

Actually... `\r?\n` is not quite working on Mac OS.. let me dig deeper.

Reply:
> > One more thing: I haven't played much with vimscript in a while, but we should maybe make sure to filter blank lines instead of only empty ones. What do you think?
> 
> What do you mean by _blank lines_? Lines with a bunch of whitespaces? I guess we could, but to be honest I focused on porting what we had before to Vimscript.

Right.

Reply:
> > I just checked the output of `goobook` and I can confirm that it's necessary to remove lines ending with `(group)` which would otherwise reflect group tags from Google contacts, including all members of the respective groups. Not sure if filtering `(other)` is necessary, though.
> 
> Just to be on the safe side, shouldn't we list them all (i.e. without additional filters) and let users figure out the exact contact they are looking for?

The thing is that these output lines can be VERY long, e.g. in case of a group with numerous members where the output would be completely garbled. For example, I have a group that consists of only 6 people, but the output line is already 264 characters wide. Also I do not think that outputting the groups would make that much sense because they are no valid recipients (at least as far as I know you cannot send anything to group labels in gmail).

Reply:
> > Correct! I have been a little too sloppy with `\r*`
> 
> Actually... `\r?\n` is not quite working on Mac OS.. let me dig deeper.

Alrighty, the `?` had to be escaped -- `\r\?\n`

Reply:
> The thing is that these output lines can be VERY long, e.g. in case of a group with numerous members where the output would be completely garbled. For example, I have a group that consists of only 6 people, but the output line is already 264 characters wide. Also I do not think that outputting the groups would make that much sense because they are no valid recipients (at least as far as I know you cannot send anything to group labels in gmail).

I see what you mean, it makes sense.

About the `(other)` thing though, I have no idea why I introduced that -- it doesn't make any sense; also, I tried running `goobook` and pipe its output into the `grep -v` command, but it did not seem like groups account were being filtered out: groups were not surrounded by braces, hence the `grep` regexp wasn't filtering them out.

Anyway, can you give the latest version a go, and see if it now works as expected?